src/domains.ts

```diff
@@ -0,0 +1,307 @@
+         * @param {string} domainID - domain ID.
+         * @param {string} token - user token.
+         * @returns {object} - returns an object domain permissions eg:
+         *  { permissions: [ 'admin', 'edit', 'view', 'membership' ] }
+         * @example
+         * const domainID = "domainID";
+         */
+    const options: RequestInit = {
+      method: 'GET',
+      headers: {
+        'Content-Type': this.contentType,
+        Authorization: `Bearer ${token}`
+      }
+    }
+
+    try {
+      const response = await fetch(
+        new URL(`${this.domainsEndpoint}/${domainID}/permissions`, this.domainsUrl).toString(),
+        options
+      )
+      if (!response.ok) {
+        const errorRes = await response.json()
+        throw this.domainError.HandleError(errorRes.error, response.status)
+      }
+      const domainData: Permissions = await response.json()
+      return domainData
+    } catch (error) {
+      throw error
+    }
+  }
+
+  public async Domains (queryParams: QueryParams, token: string): Promise<DomainsPage> {
+    // Domains retrieves all domains.
+
+    const stringParams: Record<string, string> = Object.fromEntries(
+      Object.entries(queryParams).map(([key, value]) => [key, String(value)])
+    )
+    const options: RequestInit = {
+      method: 'GET',
+      headers: {
+        'Content-Type': this.contentType,
+        Authorization: `Bearer ${token}`
+      }
+    }
+
+    try {
+      const response = await fetch(
+        new URL(`${this.domainsEndpoint}?${new URLSearchParams(stringParams).toString()}`, this.domainsUrl).toString(),
+        options
+      )
+      if (!response.ok) {
+        const errorRes = await response.json()
+        throw this.domainError.HandleError(errorRes.error, response.status)
+      }
+      const domainData = await response.json()
+      return domainData
+    } catch (error) {
+      throw error
+    }
+  }
+
+  public async ListUserDomains (userID: string, queryParams: QueryParams, token: string): Promise<DomainsPage> {
+    // ListUserDomains retrieves all domains for a user.
+
+    const stringParams: Record<string, string> = Object.fromEntries(
+      Object.entries(queryParams).map(([key, value]) => [key, String(value)])
+    )
+    const options: RequestInit = {
+      method: 'GET',
+      headers: {
+        'Content-Type': this.contentType,
+        Authorization: `Bearer ${token}`
+      }
+    }
+
+    try {
+      const response = await fetch(
+        new URL(`/users/${userID}/domains?${new URLSearchParams(stringParams).toString()}`, this.domainsUrl).toString(),
+        options
+      )
+      if (!response.ok) {
+        const errorRes = await response.json()
+        throw this.domainError.HandleError(errorRes.error, response.status)
+      }
+      const domainData = await response.json()
+      return domainData
+    } catch (error) {
+      throw error
+    }
+  }
+
+  public async EnableDomain (domainID: string, token: string): Promise<Domain> {
```

Comment:
Enable and disable domain do not return the domain. Just an status code and/or error incase of an error